client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs

```diff
@@ -95,6 +96,105 @@ public async Promise Init()
 			CheckMicroserviceStatus();
 			ConnectToLogs();
 			LogVerbose("Completed");
+			await Migrate();
+		}
+
+		public async Promise Migrate()
+		{
+			var commonCsProj = await MigrateCommon();
+			foreach(var microserviceDir in Directory.EnumerateDirectories("Assets/Beamable/Microservices"))
+			{
+				var microserviceName = Path.GetFileName(microserviceDir);
+				var path = $"{StandaloneMicroservicesPath}{microserviceName}/";
+				if (!Directory.Exists(path))
+				{
+					Debug.Log(microserviceName);
+					await CreateMicroService(microserviceName, $"{StandaloneMicroservicesPath}{microserviceName}/",
+					                         null,
+					                         true);
+				}
+
+				var signpost = _services.FirstOrDefault(s => s.name.Equals(microserviceName));
+				if(signpost == null) continue;
+				var command = $"add {signpost.CsprojFilePath} reference {commonCsProj}";

```

Comment:
When the line 112 above gets executed, wouldn't this create 2 references in the service csproj file?

Reply:
Good call